tunnel/firewall/rules.go

```diff
@@ -353,6 +356,233 @@ func permitLoopback(session uintptr, baseObjects *baseObjects, weight uint8) err
 	return nil
 }
 
+func enableSplitTunneling(session uintptr, baseObjects *baseObjects, weight uint8, splitTunnelConfig conf.SplitTunnel, tunnelLUID uint64) error {
+	calloutKey, _ := windows.GUIDFromString("{565ca03d-428a-42c9-b1a3-c8f081480f4a}")
+	_, err := createCallout(session, calloutKey, &baseObjects.splitTunnelProvider, cFWPM_LAYER_ALE_CONNECT_REDIRECT_V4, "Split-tunnel connect redirect callout")
+	if err != nil {
+		return wrapErr(err)
+	}
+
+	udpCalloutKey, _ := windows.GUIDFromString("{46f0a3fd-6404-41e3-b9f6-91b150c7ee8f}")
+	_, err = createCallout(session, udpCalloutKey, &baseObjects.splitTunnelProvider, cFWPM_LAYER_ALE_BIND_REDIRECT_V4, "Split-tunnel bind redirect callout")
+	if err != nil {
+		return wrapErr(err)
+	}
+
+	localInterface, err := getLocalInterface(tunnelLUID)
+	if err != nil {
+		return wrapErr(err)
+	}
+	providerContextKey, err := createProviderContext(session, &baseObjects.splitTunnelProvider, "Split-tunnel provider context", "Split-tunnel provider context", localInterface)
+	if err != nil {
+		return wrapErr(err)
+	}
+
+	excludedAppPaths := splitTunnelConfig.ExcludedApps
+	for _, appPath := range excludedAppPaths {
+		err = permitApp(session, baseObjects, weight, appPath)
+		if err != nil {
+			return wrapErr(err)
+		}
+
+		err = createAppFilter(session, baseObjects, cFWPM_LAYER_ALE_CONNECT_REDIRECT_V4, weight, appPath, calloutKey, providerContextKey)
+		if err != nil {
+			return wrapErr(err)
+		}
+
+		err = createAppFilter(session, baseObjects, cFWPM_LAYER_ALE_BIND_REDIRECT_V4, weight, appPath, udpCalloutKey, providerContextKey)
+		if err != nil {
+			return wrapErr(err)
+		}
+	}
+
+	return nil
+}
+
+func getLocalInterface(tunnelLUID uint64) (netip.Addr, error) {
+	interfaces, err := winipcfg.GetAdaptersAddresses(windows.AF_INET, winipcfg.GAAFlagIncludeGateways)
+	if err != nil {
+		return netip.Addr{}, err
+	}
+
+	for _, iface := range interfaces {
+		isValidInterface := iface.OperStatus == winipcfg.IfOperStatusUp && iface.FirstUnicastAddress != nil && iface.FirstGatewayAddress != nil
+		isTunnel := uint64(iface.LUID) == tunnelLUID
+
+		if isValidInterface && !isTunnel {
+			addr, ok := netip.AddrFromSlice(iface.FirstUnicastAddress.Address.IP())
+			if ok {
+				return addr, nil
+			}
+		}
+	}
+
+	return netip.Addr{}, errors.New("no valid local interface found")
```

Comment:
Just for my understanding, will we have this error written down to the local log file (log.bin)? So we can see it in metadata.